src/sanity_checker.rs

```diff
@@ -137,84 +138,80 @@ pub fn check_config_option_sanity(config_options: &Input) -> Result<(), String>
     }
 }
 
-fn check_ports_are_valid(option_ports: &Option<Vec<String>>, name: &str) -> bool {
+fn check_ports_are_valid(option_ports: &Option<Vec<u16>>, name: &str) -> bool {
     let mut is_input_sane = true;
     // if we have a zero length vector the input is always bad
-    match option_ports {
-        Some(ports) => {
-            if ports.is_empty() {
-                return false;
-            }
-
-            // if the vector value of the first element is zero length the user
-            // didn't specify the option and the input is valid
-            if ports[0].is_empty() {
-                return true;
-            }
-
-            for port in ports {
-                match port.chars().all(char::is_numeric)
-                    && port.parse().unwrap_or(0) > 0
-                    && port.parse().unwrap_or(65536) < 65535
-                {
-                    true => trace!("{} UDP Port is numeric. Found: {}", name, port),
-                    false => {
-                        error!(
-                            "{} UDP Listen Port is not numeric or out of the range of 1-65353. Found: {}",
-                            name, port
-                        );
-                        is_input_sane = false;
-                    }
-                }
+    if let Some(ports) = option_ports {
```

Comment:
Changed from a `match` block to use an `if let Some()` as you don't need to handle `None`.